electrolytes/data2lammps.py

```diff
@@ -0,0 +1,326 @@
+  {-boxsize/2} {boxsize/2} zlo zhi
+}}
+    """
+    f = open(directory+f"/{filename}.lt","w")
+    f.write(systemlt)
+    f.close()
+
+    # Given the system LT and PDB file, which is generated from Packmol, run Moltemplate
+    bashCommand = f"cd {directory}; moltemplate.sh -pdb {filename}.pdb {filename}.lt; cd ..;"
+    os.system(bashCommand)
+    
+    # Cleanup the files generated from moltemplate
+    bashCommand = f"cd {directory}; cleanup_moltemplate.sh -base {filename}; cd ..;"
+    os.system(bashCommand)
+    
+    # Next check the settings file and see if LAMMPS hybrid style is even necessarry
+    # Begin by reading the entire contents of the *in.settings file
+    file = open(directory+f'/{filename}.in.settings', 'r')
+    lines = file.readlines()
+    string = ""
+    for i, line in enumerate(lines):
+        if "dihedral_coeff" in line:
+            print(line)
+            if 'opls' or 'fourier' in line:
+                string += line.split()[2]
+    file.close()
+    
+    if 'opls' in string and 'fourier' in string:
+        print("No modification necesssarry. Hybrid style is present")
+    else:
+        print("Modification necesssarry. Only either opls/fourier style present")
+        
+        #First we modify the lines containing dihedral_coeff and delete the keyword for opls and fourier
+        for i, line in enumerate(lines):
+            if "dihedral_coeff" in line:
+                if 'opls' or 'fourier' in line:
+                    modified_line = line.replace('opls', '')
+                    modified_line = modified_line.replace('fourier', '')
+                    lines[i] = modified_line
+        with open(directory+f'/{filename}.in.settings', 'w') as file:
+            file.writelines(lines)
+        
+        # Next, modify the *in.init file. We need to remove the hybrid keyword from that file.
+        with open(directory+f'/{filename}.in.init', 'r') as file:
+            lines = file.readlines()
+        for i, line in enumerate(lines):
+            if "dihedral_style" in line:
+                if 'opls' or 'fourier' in line:
+                    modified_line = line.replace('hybrid', '')
+                    if 'opls' in string:
+                        modified_line = modified_line.replace('fourier', '')
+                    if 'fourier' in string:
+                        modified_line = modified_line.replace('opls', '')
+                    lines[i] = modified_line
+        with open(directory+f'/{filename}.in.init', 'w') as file:
+            file.writelines(lines)
+
+def extract_elements_and_counts(formula):
+    """ Return the elements and stoichiometry of a chemical species, given its formula
+        
+        Args:
+            formula (string): a chemical formula, which may include their charge. 
+            Example: ethanol C2H6O or hexafluorophosphate F6P-
+    """
+    # Split the formula based on + or - signs and take only the part before it
+    formula = formula.split('+')[0].split('-')[0]
```

Comment:
I'm doing it this way because I am not expecting the formula names to be formatted that way, i.e., +/- are strictly information about charge. 

What I should do is probably catch this scenario (P-F6) and say that this is wrong way of formatting. After - or + sign cannot be a letter.